mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,360 @@
+        If reject is None then no rejection is done. Example::
+
+            reject = dict(grad=4000e-13, # T / m (gradiometers)
+                          mag=4e-12, # T (magnetometers)
+                          eeg=40e-6, # uV (EEG channels)
+                          eog=250e-6 # uV (EOG channels)
+                          )
+    flat : dict | None
+        Rejection parameters based on flatness of signal.
+        Valid keys are 'grad' | 'mag' | 'eeg' | 'eog' | 'ecg', and values
+        are floats that set the minimum acceptable peak-to-peak amplitude.
+        If flat is None then no rejection is done.
+    reject_tmin : scalar | None
+        Start of the time window used to reject epochs (with the default None,
+        the window will start with tmin).
+    reject_tmax : scalar | None
+        End of the time window used to reject epochs (with the default None,
+        the window will end with tmax).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Notes
+    -----
+    .. versionadded:: 0.11.0
+
+    See Also
+    --------
+    mne.Epochs : Documentation of attribute and methods.
+    """
+    @verbose
+    def __init__(self, input_fname, events=None, event_id=None, tmin=0,
+                 baseline=None,  reject=None, flat=None, reject_tmin=None,
+                 reject_tmax=None, montage=None, verbose=None):
+        from scipy import io
+        eeg = io.loadmat(input_fname, struct_as_record=False,
+                         squeeze_me=True)['EEG']
+
+        if events is None and eeg.trials > 1:
+            # first extract the events and construct an event_id dict
+            event_type, event_latencies, unique_ev = [], [], []
+            for ep in eeg.epoch:
+                if not isinstance(ep.eventtype, string_types):
+                    raise ValueError('An epoch can have only one event'
+                                     ' in mne-python')
+                else:
+                    event_type.append(ep.eventtype)
+                    event_latencies.append(ep.eventurevent)
+                    if ep.eventtype not in unique_ev:
+                        unique_ev.append(ep.eventtype)
+                event_id = dict((ev, idx) for idx, ev in enumerate(unique_ev))
+            # now fill up the event array
+            events = np.zeros((eeg.trials, 3), dtype=int)
+            for idx in range(eeg.trials):
+                events[idx, 0] = event_latencies[idx]
+                events[idx, 1:] = event_id[event_type[idx]]
+        elif isinstance(events, string_types):
+            events = read_events(events)
+
+        logger.info('Extracting parameters from %s...' % input_fname)
+        input_fname = op.abspath(input_fname)
+        info = _get_info(eeg, montage)
+
+        if event_id is None:  # convert to int to make typing-checks happy
+            event_id = dict((ev, idx) for idx, ev in enumerate(unique_ev))
```

Comment:
I don't think the logic here is safe. If `events is None`, `event_id is None`, and `n_trials = 1`, you will hit an error on this line because `unique_ev` doesn't exist. You will also hit an error if `events` is a `string_types` but `event_id is None`, again because `unique_ev` will not exist. Can you refactor the conditionals, and also add tests that hit all such corner cases?